src/planck_integral.hpp

```diff
@@ -0,0 +1,241 @@
+//==============================================================================
+//==============================================================================
+/// \file planck_integral.hpp
+/// \brief Some functions for quickly integrating the Planck function.
+///
+
+#ifndef PLANCKINTEGRAL_HPP_ // NOLINT
+#define PLANCKINTEGRAL_HPP_
+
+#include <algorithm>
+#include <cmath>
+
+#include "AMReX.H"
+#include "AMReX_Array.H"
+#include "AMReX_BLassert.H"
+#include "AMReX_Extension.H"
+#include "AMReX_GpuQualifiers.H"
+#include "AMReX_REAL.H"
+
+#include "valarray.hpp"
+
+using Real = amrex::Real;
+
+static constexpr double PI = 3.14159265358979323846;
```

Comment:
Although it is not part of the standard, can you use `M_PI` instead (https://www.gnu.org/software/libc/manual/html_node/Mathematical-Constants.html)?

Reply:
Once we switch to C++20, we can use `std::numbers::pi`: https://en.cppreference.com/w/cpp/numeric/constants